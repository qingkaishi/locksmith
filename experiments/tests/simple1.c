#include <pthread.h>

pthread_mutex_t lock;
int shared;

void *run(void *arg) {
  pthread_mutex_lock(&lock);
  shared++;
  printf("foo");
  pthread_mutex_unlock(&lock);
  printf("foo");
  return NULL;
}
int main() {
  pthread_t t1, t2;

  pthread_mutex_init(&lock, NULL);
  pthread_create(&t1, NULL, run, NULL);
  printf("foo");
  pthread_create(&t2, NULL, run, NULL);
  return 1;
}
