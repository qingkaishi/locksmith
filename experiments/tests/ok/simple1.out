************************* STARTING *************************


simple1.c:9: Warning: assigning pointer to non-pointer: int*^{const-string:simple1.c:9}
 ->
 void

simple1.c:11: Warning: assigning pointer to non-pointer: int*^{const-string:simple1.c:11}
 ->
 void

simple1.c:19: Warning: assigning pointer to non-pointer: int*^{const-string:simple1.c:19}
 ->
 void
functions declared and used but not defined:
  printf
shared: &shared:simple1.c:4
&shared:simple1.c:4 is protected by:
  concrete lock:simple1.c:17
  lock:simple1.c:3
*************************** DONE ***************************
